pkg/ibmvpc/implicit_routing.go

```diff
@@ -83,14 +104,24 @@ func pgwHasSource(src vpcmodel.Node, pgw *PublicGateway) bool {
 	// another option: compare by nodes within pgw.src (currently breaks test)
 }
 
-// getPath returns a path from src to dst if such exists, or nil otherwise
+// getIngressPath returns a path to dest
+func (rt *systemImplicitRT) getIngressPath(dest *ipblock.IPBlock) (vpcmodel.Path, error) {
+	// traffic from some source is by default simply routed to dest node
+	path := rt.destAsPath(dest)
+	if path == nil {
+		return nil, fmt.Errorf("getIngressPath: failed to find path to dest resource address %s in VPC %s", dest.String(), rt.vpc.Name())
+	}
+	return path, nil
+}
+
+// getEgressPath returns a path from src to dst if such exists, or nil otherwise
 // TODO: src should be InternalNodeIntf, but it does not implement VPCResourceIntf
-func (rt *systemImplicitRT) getPath(src vpcmodel.Node, dest *ipblock.IPBlock) vpcmodel.Path {
+func (rt *systemImplicitRT) getEgressPath(src vpcmodel.Node, dest *ipblock.IPBlock) vpcmodel.Path {
 	// TODO: split dest by disjoint ip-blocks of the vpc-config (the known destinations ip-blocks)
 
 	if dest.ContainedIn(rt.vpc.AddressPrefixes()) {
 		// direct connection
-		return []*vpcmodel.Endpoint{{VpcResource: src}, {IPBlock: dest}}
+		return vpcmodel.ConcatPaths(vpcmodel.PathFromResource(src), rt.destAsPath(dest))
```

Comment:
Make this change throughout the function. Otherwise it is confusing.

Reply:
done